library/src/main/java/com/mapzen/android/MapzenMap.java

```diff
@@ -676,4 +708,26 @@ public void drawTransitRouteLine(@NotNull List<LngLat> points, @Nullable List<Ln
   public void clearTransitRouteLine() {
     overlayManager.clearTransitRouteLine();
   }
+
+  /**
+   * By default all {@link MapData} objects are removed from the map when it is destroyed. To
+   * persist this data such as in the case of orientation changes, use this method.
+   * @param persistOnRecreation persist {@link MapData} across orientation changes
+   */
+  public void setPersistMapData(boolean persistOnRecreation) {
+    overlayManager.setPersistMapData(persistOnRecreation);
+    mapStateManager.setPersistMapState(persistOnRecreation);
+  }
+
+  private void restoreMapState() {
```

Comment:
Really nice catch. I updated the code so that we only set the scene file once, in the `MapInitializer`